src/main/java/io/vavr/collection/Iterator.java

```diff
@@ -1033,7 +1033,11 @@ public T next() {
     // -- Additional methods of Iterator
 
     @Override
+    @SuppressWarnings("unchecked")
     default <R> Iterator<R> collect(PartialFunction<? super T, ? extends R> partialFunction) {
+        if (!hasNext()) {
```

Comment:
I'm not sure if this is adheres to coding guidelines.
If `partialFunction == null` should not throw when empty, this additional check is required, because the expression partialFunction::isDefinedAt will otherwise throw immediately.